tsconfig.json

```diff
@@ -10,6 +10,7 @@
     "resolveJsonModule": true,
     "declaration": true,
     "declarationDir": "lib/types",
+    "skipLibCheck": true,
```

Comment:
Skip checking `node_modules`, there's a problem with it because we include `dom` and `webworker`, both have identical types that clash. For example:

```
node_modules/typescript/lib/lib.webworker.d.ts:5951:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'onrejectionhandled' must be of type '((this: Window, ev: PromiseRejectionEvent) => any) | null', but here has type '((this: DedicatedWorkerGlobalScope, ev: PromiseRejectionEvent) => any) | null'.

5951 declare var onrejectionhandled: ((this: DedicatedWorkerGlobalScope, ev: PromiseRejectionEvent) => any) | null;
                 ~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:19933:13
    19933 declare var onrejectionhandled: ((this: Window, ev: PromiseRejectionEvent) => any) | null;
                      ~~~~~~~~~~~~~~~~~~
    'onrejectionhandled' was also declared here.
```

Reply:
Doesn't `skipLibCheck: true` negates all type checking originating from third-party libraries?

Can you please verify that we still get type violations if we use some third-party package against its defined types?

Reply:
You're right @kettanaito, third party type violations will not be caught. 
[See docs](https://www.typescriptlang.org/docs/handbook/release-notes/overview.html#new---skiplibcheck)

If we want to keep this enabled, I think the only solution would be to create multiple `tsconfig`'s.
One for each build, so they can have their own included libs.
This way they won't interfere with eachother.

Reply:
TS configs per each build target sound like a sensible idea. We won't have to compose TS configs in `rollup.config.ts`.

Reply:
Adding those declaration files to our excludes doesn't help.
How do you want to continue with this? Should we try a setup with multiple `tsconfig` files?

Reply:
Let's try multiple `tsconfig.json` configurations. If you feel it starts to get messy and takes too much time, we can always discuss this and find a different solution. Thank you for your investigation on this, by the way!

Reply:
I'm confused.

I thought it wouldn't be able to add multiple `tsconfig`s because they are both (dom and webworker) included in our bundle (both are exported via the index file). Now the weird part is that I was able to exclude the webworker types after I modified some types in 2b59cd94f5894f2389588408a28534a5dd8cc51e. 

After this, everything worked, but I don't exactly know why...